internal/storage/bucket_handle.go

```diff
@@ -72,6 +73,26 @@ func (b *bucketHandle) DeleteObject(ctx context.Context, req *gcs.DeleteObjectRe
 		obj = obj.If(storage.Conditions{MetagenerationMatch: *req.MetaGenerationPrecondition})
 	}
 
-	// Deleting object through Go Storage Client.
 	return obj.Delete(ctx)
 }
+
+func (b *bucketHandle) StatObject(ctx context.Context, req *gcs.StatObjectRequest) (o *gcs.Object, err error) {
+	var attrs *storage.ObjectAttrs
+	// Retrieving object attrs through Go Storage Client.
+	attrs, err = b.bucket.Object(req.Name).Attrs(ctx)
+
+	// If error is of type storage.ErrObjectNotExist
+	if err == storage.ErrObjectNotExist {
+		err = &gcs.NotFoundError{Err: err} // Special case error that object not found in the bucket.
```

Comment:
instead we should directly return storage.ErrObjectNotExist error and handle it whereever gcs.NotFoundError is used in gcsFuse. You can do that chages in  followup cl

Reply:
Since we anyways have dependency on request/response protos, keep it like this for now.

Check if you can add a test case to verify if gcs.NotFoundError is return when ErrObjectNnotExist is returned.

Reply:
Yes it is returning

Reply:
Yes it is returning